### PR TITLE
Adds `track` option to fine-tune which actions must be stored in the history

### LINKED ISSUE #262: Feature: ignore certain actions in the reducer
h2. Context

We work on a drawing tool where users can draw polygons on a map. They can also reposition the vertices of the drawn polygon. A single reducer manages every action.

h2. Problem

When the user moves a vertex, we constantly send actions to the reducer to update the state and rerender the image. At the moment, `react-undo-redo` captures every such action. When users try to undo their action, they do not see the change because it is too small. Only by holding the undo button they can see how the vertex moves back to the previous position

h2. Proposal

Introduce a way to ignore actions and pass them straight to the reducer without putting their state to the undo/redo state.

Possible API:
```ts
const { UndoRedoProvider, usePast, usePresent, useFuture, useUndoRedo } =
  createUndoRedo(reducer, { ignoreActions: ['UPDATE_POINT', 'NON_VISUAL_CHANGE'] })
```

### PR DESCRIPTION
**Pull request by nlopin**
Adds second `options` parameter to `createUndoRedo` and `createReducer`. The only available option key for now is `track` which takes a predicate. I've gone with this solution to stay flexible (for example, in case we'll decide to provide an extra `ignoreActions` option)

Closes #262

### CODE REVIEW

File: src/createReducer.ts [hunk 1]
```diff
@@ -60,6 +67,15 @@ export function createReducer<Present, Actions extends {}>(
       }
     }
 
+    const isTrackableAction = typeof options?.track === 'function' ? options.track(action) : trackAll()
```
Comment (nlopin) [inline src/createReducer.ts:70]: Do you think the way I manage optional value is okay for you? 

The alternative was `options?.track?.(action) ?? trackAll()`, but this way we'll pass the control to the `trackAll` function if `track` returns `undefined` (for example, if people use JS). I'd rather consider it as `false`. 

Feel free to let me know if you have any doubts. Maybe I'm overthinking the solution here.

Comment (nlopin) [reply src/createReducer.ts]: After consideration, I decided that a better API would be to consider only explicit `false` as a way to skip the action. In that approach `undefined` means — use the default, which is `true`

**Comment by nlopin** [comment]
Everything is ready for the review @frontendphil

**Comment by frontendphil** [comment]
Sorry. I was reviewing this and then GitHub had issues.

File: src/createReducer.ts [hunk 2]
```diff
@@ -29,8 +33,11 @@ type RedoAction = {
 
 export type UndoRedoActions<Base> = Base | UndoAction | RedoAction
 
+const trackAll = () => true
+
 export function createReducer<Present, Actions extends {}>(
-  presentReducer: PresentReducer<Present, Actions>
+  presentReducer: PresentReducer<Present, Actions>,
+  options: UndoRedoOptions<Actions> = {}
```
Comment (frontendphil) [inline src/createReducer.ts:40]:
```suggestion
  { track = trackAll }: UndoRedoOptions<Actions> = {}
```

File: src/createReducer.ts [hunk 3]
```diff
@@ -60,6 +67,15 @@ export function createReducer<Present, Actions extends {}>(
       }
     }
 
+    const isTrackableAction = options?.track?.(action) ?? trackAll()
```
Comment (frontendphil) [inline src/createReducer.ts:70]:
```suggestion
    const isTrackableAction = track(action)
```

**Review by frontendphil — APPROVED** [review]
Hey @nlopin I've left two suggestions in the code. I, personally, like to avoid optional chaining paired with nullish coalescing. Makes me think so much. If you agree with my suggestions feel free to incorporate them and we can merge this right after. I've approved nonetheless. So you can also go a head and merge the changes as-is.

**Comment by nlopin** [comment]
Great suggestions @frontendphil, I like the simplification. Unfortunately, I cannot merge by myself because it requires write permissions to the repo

**Comment by nlopin** [comment]
@frontendphil is there anything I can do to merge it faster?

**Pull request merged**